platform/openide.loaders/src/org/openide/loaders/TemplateWizard.java

```diff
@@ -244,7 +252,14 @@ public DataFolder getTemplatesFolder () {
     public DataFolder getTargetFolder () throws IOException {
         LOG.log(Level.FINE, "targetFolder={0} for {1}", new Object[] {targetDataFolder, this});
         if (targetDataFolder == null) {
-            throw new IOException(NbBundle.getMessage(TemplateWizard.class, "ERR_NoFilesystem"));
+            // Fix for [NETBEANS-3918] and [NETBEANS-3875]
+            File targetFolderFile = (File) getProperty(PROP_TARGET_FOLDER_HINT);
```

Comment:
The proposed property `PROP_TARGET_FOLDR_HINT` can hold any value, not just `File`. To prevent `ClassCastException` change this and next line to `if (value instanceof File) { ... }`.